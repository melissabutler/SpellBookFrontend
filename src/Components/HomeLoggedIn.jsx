import React from "react";

const HomeLoggedIn = ({user}) => {
    return (
        <div className="Home-login">
            <h3>Welcome back, {user.username}</h3>
            
        </div>
    )
}

export default HomeLoggedIn;