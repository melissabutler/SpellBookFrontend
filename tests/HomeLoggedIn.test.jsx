import React, { useContext } from "react";
import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import HomeLoggedIn from "../src/Components/HomeLoggedIn";

describe(" HomeLoggedIn", function() {

    vi.mock('react', async (importOriginal) => {
        const react = await importOriginal('react');
        return {
            ...react,
            useContext: () => ({ currentUser: {
                username: 'u1',
                email: 'email@email.com',
                isAdmin: false,
                characters: [
                    {
                    char_class: "cleric",
                    char_name: "Beebo",
                    id: 1,
                    lvl: 3
                },
                ]
            }})
    }})


    it("renders without crashing", async function() {
        render(<HomeLoggedIn user={{username: 'u1',
            email: 'email@email.com',
            isAdmin: false,
            characters: [
                {
                char_class: "cleric",
                char_name: "Beebo",
                id: 1,
                lvl: 3
            },
            ]}}/>)
    })

    it("matches snapshot", function() {
        const { asFragment } = render(<HomeLoggedIn user={{username: 'u1',
            email: 'email@email.com',
            isAdmin: false,
            characters: [
                {
                char_class: "cleric",
                char_name: "Beebo",
                id: 1,
                lvl: 3
            },
            ]}} />);
        expect(asFragment()).toMatchSnapshot();
    })



})
