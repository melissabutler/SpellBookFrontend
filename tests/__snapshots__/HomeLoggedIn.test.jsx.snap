// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` HomeLoggedIn > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home-login"
  >
    <h3>
      Welcome back, u1
    </h3>
  </div>
</DocumentFragment>
`;
